picard/ui/itemviews.py

```diff
@@ -784,10 +785,12 @@ def dropMimeData(self, parent, index, data, action):
         # application/picard.album-list
         albums = data.data("application/picard.album-list")
         if albums:
-            albums = [self.tagger.load_album(id) for id in bytes(albums).decode().split("\n")]
+            album_ids = bytes(albums).decode().split("\n")
+            log.debug("Dropped albums = %r", album_ids)
+            files = list(iter_files_from_objects(self.tagger.load_album(id) for id in album_ids))
```

Comment:
I think we don't need the list conversion here, passing on the iterator would be ok. Especially because this still runs on the main thread later on in tagger.move_files.

The advantage would be that we move even more work out of the drop handler

Reply:
Done & rebased in 8f90884